assets/js/googlesitekit/datastore/user/feature-tours.js

```diff
@@ -283,6 +340,40 @@ const baseSelectors = {
 
 		return dismissedTourSlugs.includes( slug );
 	} ),
+
+	/**
+	 * Gets the timestamp for the last dismissed feature tour.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {number} Timestamp of the last dismissal.
+	 */
+	getLastDismissedAt( state ) {
+		return state.lastDismissedAt;
+	},
+
+	/**
+	 * Determines whether feature tours are on cooldown (i.e. the last
+	 * dismissal was within the cooldown time span).
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {undefined|boolean} Whether feature tours are on cooldown or undefined.
+	 */
+	areFeatureToursOnCooldown: ( createRegistrySelector( ( select ) => () => {
+		const lastDismissedAt = select( STORE_NAME ).getLastDismissedAt();
+
+		if ( undefined === lastDismissedAt ) {
+			return undefined;
+		}
```

Comment:
`getLastDismissedAt()` has a resolver that sets `lastDismissedAt` only if it has a truthy value. But if nothing is obtained, `lastDismissedAt` isn't set as `false`, it simply isn't set resulting in it being `undefined`.

This means, from what I can see, it'll never be set as `false`.

Reply:
But `lastDismissedAt` might be set and have a timestamp from more than 2 hours ago, which would mean this does return `false`? 🤔 

| `lastDismissedAt`  timestamp    | Function returns |
| ----------- | ----------- |
| Over 2 hours ago      | `false`       |
| Under 2 hours ago   | `true`        |
| `undefined`   | `undefined`        |

Maybe I have misunderstood?

Reply:
Looks right to me 👍

Reply:
Actually I think what @tofumatt was getting at here is that the selector doesn't differentiate between a resolving state (`undefined`) and a cache miss which would result in nothing being received so the state would be the same as resolving. I've updated this to receive `null` for the cache miss so this is always called, and then this selector would always return `false` in that case. This also fixed a bug where it was calling `setLastDismissedAt` instead of `receive*` from the resolver which was resulting in unnecessary calls to set the cache.